packages/ckeditor5-font/src/utils.ts

```diff
@@ -140,3 +147,79 @@ function normalizeColorCode( value: string ): string {
 export type ColorTableDropdownView = DropdownView & {
 	colorTableView?: ColorTableView;
 };
+
+/**
+ * Parses and converts the color string to requested format. Handles variety of color spaces
+ * like `hsl`, `hex` or `rgb`.
+ *
+ * @param color
+ * @returns A color string.
+ */
+export function convertColor( color: string, outputFormat: ColorPickerOutputFormat ): string {
+	if ( !color ) {
+		return '';
+	}
+
+	// Parser library treats `hex` format as belonging to `rgb` space, which messes up further conversion.
+	// Let's parse such strings on our own.
+	const colorObject = color.startsWith( '#' ) ? { space: 'hex', values: color.substring( 1 ) } : parse( color );
+
+	if ( !colorObject.space ) {
+		return '';
+	}
+
+	if ( colorObject.space === outputFormat ) {
+		return color;
+	}
+
+	// @ts-ignore
+	const convertedColorChannels: Array<number> = convert[ colorObject.space ][ outputFormat ]( colorObject.values );
+
+	return formatColorOutput( convertedColorChannels, outputFormat );
+}
+
+/**
+ * Color formats handled by color converter.
+ */
+export type ColorPickerOutputFormat = 'hex' | 'rgb' | 'hsl' | 'hwb' | 'lab' | 'lch';
```

Comment:
There's `color` format missing, but we can skip it for now to simplify implementation.

Reply:
I'm not sure if you mean [the color notation](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color) (e.g. `color(display-p3 1 0.5 0);`) here or using keywords (like `blue`). The first one is not supported neither by `color-convert` nor `color-parse` libs, so it doesn't seem likely to include it in the list. Keywords, on the other hand, could be a valid inclusion, but in picker -> output conversion they would only be rounded to the closest keyword, so I'm not sure how useful it would be. Anyway, I extracted the follow-up here: cksource/ckeditor5-internal#3046.

Reply:
That's about later. Let's keep it in a MVP1234 follow-up.